service/Abstractions/ContentStorage/IContentStorage.cs

```diff
@@ -66,13 +66,15 @@ public interface IContentStorage
     /// <param name="index">Index name</param>
     /// <param name="documentId">Document ID</param>
     /// <param name="fileName">Name of the file</param>
+    /// <param name="contentType">Mime type of the file</param>
     /// <param name="streamContent">File content</param>
     /// <param name="cancellationToken">Async task cancellation token</param>
     Task WriteFileAsync(
         string index,
         string documentId,
         string fileName,
         Stream streamContent,
+        string contentType = "application/octet-stream",
```

Comment:
I would move this change to a separate PR to be reviewed and merged first. By the way, any change to Abstractions needs to be done separately due to how nugets are wired up.
Something like this:

PR1: changes to IContentStorage
PR2: changes to classes implementing IContentStorage
PR3: rest of the PR (we'll have to split again probably but for now we can do all in one smaller PR)